### ISSUE TITLE
Compilation Error when Installing with Cabal and Pandoc 2.1.3

### ISSUE THREAD

**Issue by rriemann** [mb21/pandoc-placetable#20]
Hello,

I tried to setup pandoc-placetable in a Debian-based docker.

```Dockerfile
FROM debian:sid
RUN apt-get update
RUN apt-get install -y --no-install-recommends zlib1g-dev unzip zip ghc cabal-install dnsutils
RUN cabal update
RUN cabal install -f inlineMarkdown pandoc-placetable pandoc-crossref pandoc-citeproc
```

The last command `cabal install`gives the following errors:

```
...
Downloading pandoc-2.1.3...
Configuring pandoc-2.1.3...
Building pandoc-2.1.3...
Installed pandoc-2.1.3
Downloading pandoc-placetable-0.4.1...
Downloading pandoc-crossref-0.3.1.0...
Downloading pandoc-citeproc-0.14.3.1...
Configuring pandoc-placetable-0.4.1...
Configuring pandoc-crossref-0.3.1.0...
Building pandoc-placetable-0.4.1...
Building pandoc-crossref-0.3.1.0...
Failed to install pandoc-placetable-0.4.1
Build log ( /root/.cabal/logs/ghc-8.2.2/pandoc-placetable-0.4.1-AMD8DdWHNCv8O88m4uAws4.log ):
cabal: Entering directory '/tmp/cabal-tmp-6/pandoc-placetable-0.4.1'
Configuring pandoc-placetable-0.4.1...
Preprocessing executable 'pandoc-placetable' for pandoc-placetable-0.4.1..
Building executable 'pandoc-placetable' for pandoc-placetable-0.4.1..

<no location info>: warning: [-Wmissing-home-modules]
    These modules are needed for compilation but not listed in your .cabal file's other-modules: Paths_pandoc_placetable
[1 of 2] Compiling Paths_pandoc_placetable ( dist/build/pandoc-placetable/autogen/Paths_pandoc_placetable.hs, dist/build/pandoc-placetable/pandoc-placetable-tmp/Paths_pandoc_placetable.o )
[2 of 2] Compiling Main             ( pandoc-placetable.hs, dist/build/pandoc-placetable/pandoc-placetable-tmp/Main.o )

pandoc-placetable.hs:110:59: error:
    * Couldn't match type `[Char]'
                     with `text-1.2.3.0:Data.Text.Internal.Text'
      Expected type: [text-1.2.3.0:Data.Text.Internal.Text]
        Actual type: [String]
    * In the second argument of `map', namely `headers'
      In the third argument of `table', namely
        `(map strToBlocks headers)'
      In the expression:
        table (strToInlines caption) cellspecs (map strToBlocks headers)
    |
110 |   table (strToInlines caption) cellspecs (map strToBlocks headers)
    |                                                           ^^^^^^^

pandoc-placetable.hs:111:31: error:
    * Couldn't match type `[Char]'
                     with `text-1.2.3.0:Data.Text.Internal.Text'
      Expected type: [[text-1.2.3.0:Data.Text.Internal.Text]]
        Actual type: [[String]]
    * In the second argument of `map . map', namely `rows'
      In the second argument of `($)', namely
        `(map . map) strToBlocks rows'
      In the expression:
        table (strToInlines caption) cellspecs (map strToBlocks headers)
          $ (map . map) strToBlocks rows
    |
111 |     $ (map . map) strToBlocks rows
    |                               ^^^^

pandoc-placetable.hs:136:38: error:
    * Couldn't match expected type `text-1.2.3.0:Data.Text.Internal.Text'
                  with actual type `[Char]'
    * In the second argument of `readMarkdown', namely s'
      In the expression: readMarkdown def s'
      In the expression:
        case readMarkdown def s' of
          Right (Pandoc _ bs) -> fromList $ extractIns $ head bs
          Left e -> str $ show e
    |
136 |            in  case readMarkdown def s' of
    |                                      ^^

pandoc-placetable.hs:151:24: error:
    * Couldn't match type `text-1.2.3.0:Data.Text.Internal.Text'
                     with `[Char]'
      Expected type: String
        Actual type: text-1.2.3.0:Data.Text.Internal.Text
    * In the first argument of `str', namely `s'
      In the second argument of `($)', namely `str s'
      In the expression: plain $ str s
    |
151 |            plain $ str s
    |                        ^
cabal: Leaving directory '/tmp/cabal-tmp-6/pandoc-placetable-0.4.1'
Configuring pandoc-citeproc-0.14.3.1...
Building pandoc-citeproc-0.14.3.1...
Installed pandoc-crossref-0.3.1.0
Installed pandoc-citeproc-0.14.3.1
cabal: Error: some packages failed to install:
pandoc-placetable-0.4.1-AMD8DdWHNCv8O88m4uAws4 failed during the building
phase. The exception was:
ExitFailure 1
The command '/bin/sh -c cabal install -f inlineMarkdown pandoc-placetable pandoc-crossref pandoc-citeproc' returned a non-zero code: 1
```

**Comment by mb21** [comment]
I've just pushed a commit to make it compile with pandoc 2.x... can you try current master branch?

**Comment by mb21** [comment]
I just released pandoc-placetable 0.5 with those changes.

**Comment by rriemann** [comment]
Thank you so much, it worked!
